src/main/java/bio/terra/service/JobService.java

```diff
@@ -217,7 +234,21 @@ public JobModel retrieveJob(String jobId, AuthenticatedUserRequest userReq) {
         String jobId,
         Class<T> resultClass,
         AuthenticatedUserRequest userReq) {
-        stairway.verifyListFlightAccess(jobId, buildUserRequestInfo(userReq));
+        boolean canListAnyJob = true;
+        if (userReq != null) {
+            // currently, this check will be true for stewards only
+            canListAnyJob = samService.isAuthorized(
+                userReq,
+                SamClientService.ResourceType.DATAREPO,
+                appConfig.getResourceId(),
+                SamClientService.DataRepoAction.LIST_JOBS);
+        }
```

Comment:
You might make that check a method. The code looks the same in each of the methods (I think...)